### ISSUE TITLE
[colmap] build failure: Could not find FreeImage

### ISSUE THREAD

**Issue by ThGravo** [microsoft/vcpkg#14905]
**Host Environment**
- OS: Ubuntu 20.04
- Compiler: gcc version 9.3.0 

**To Reproduce**
`./vcpkg install colmap`

**Failure logs** 
```
Building package colmap[core]:x64-linux...
Could not locate cached archive: /home/fgawrilo/.cache/vcpkg/archives/66/66bce329529a8a2c261e55bff4b248f9d875196b.zip
-- Using cached /media/data/Code/repos/3rd/vcpkg/downloads/colmap-colmap-3.6.tar.gz
-- Cleaning sources at /media/data/Code/repos/3rd/vcpkg/buildtrees/colmap/src/3.6-30c64fd4c7.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /media/data/Code/repos/3rd/vcpkg/downloads/colmap-colmap-3.6.tar.gz
-- Using source at /media/data/Code/repos/3rd/vcpkg/buildtrees/colmap/src/3.6-30c64fd4c7.clean
-- Configuring x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:108 (message):
    Command failed: /media/data/Code/repos/3rd/vcpkg/downloads/tools/cmake-3.18.4-linux/cmake-3.18.4-Linux-x86_64/bin/cmake /media/data/Code/repos/3rd/vcpkg/buildtrees/colmap/src/3.6-30c64fd4c7.clean -DCUDA_ENABLED=OFF -DCUDA_ARCHS= -DTESTS_ENABLED=OFF -DGIT_COMMIT_ID=3.6 -DGIT_COMMIT_DATE=2020-12-02 -DCMAKE_MAKE_PROGRAM=/home/fgawrilo/repos/3rd/vcpkg/downloads/tools/ninja-1.10.1-linux/ninja -DCMAKE_SYSTEM_NAME=Linux -DBUILD_SHARED_LIBS=OFF -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/home/fgawrilo/repos/3rd/vcpkg/scripts/toolchains/linux.cmake -DVCPKG_TARGET_TRIPLET=x64-linux -DVCPKG_SET_CHARSET_FLAG=ON -DVCPKG_PLATFORM_TOOLSET=external -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE -DCMAKE_VERBOSE_MAKEFILE=ON -DVCPKG_APPLOCAL_DEPS=OFF -DCMAKE_TOOLCHAIN_FILE=/home/fgawrilo/repos/3rd/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON -DVCPKG_CXX_FLAGS= -DVCPKG_CXX_FLAGS_RELEASE= -DVCPKG_CXX_FLAGS_DEBUG= -DVCPKG_C_FLAGS= -DVCPKG_C_FLAGS_RELEASE= -DVCPKG_C_FLAGS_DEBUG= -DVCPKG_CRT_LINKAGE=dynamic -DVCPKG_LINKER_FLAGS= -DVCPKG_LINKER_FLAGS_RELEASE= -DVCPKG_LINKER_FLAGS_DEBUG= -DVCPKG_TARGET_ARCHITECTURE=x64 -DCMAKE_INSTALL_LIBDIR:STRING=lib -DCMAKE_INSTALL_BINDIR:STRING=bin -D_VCPKG_ROOT_DIR=/media/data/Code/repos/3rd/vcpkg -D_VCPKG_INSTALLED_DIR=/media/data/Code/repos/3rd/vcpkg/installed -DVCPKG_MANIFEST_INSTALL=OFF -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/media/data/Code/repos/3rd/vcpkg/packages/colmap_x64-linux/debug
    Working Directory: /media/data/Code/repos/3rd/vcpkg/buildtrees/colmap/x64-linux-dbg
    Error code: 1
    See logs for more information:
      /media/data/Code/repos/3rd/vcpkg/buildtrees/colmap/config-x64-linux-dbg-out.log
      /media/data/Code/repos/3rd/vcpkg/buildtrees/colmap/config-x64-linux-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:328 (vcpkg_execute_required_process)
  ports/colmap/portfile.cmake:40 (vcpkg_configure_cmake)
  scripts/ports.cmake:136 (include)


Error: Building package colmap:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `./vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: colmap:x64-linux
  Vcpkg version: 2020.11.12-unknownhash

Additionally, attach any relevant sections from the log files above.
```
[config-x64-linux-dbg-err.log](https://github.com/microsoft/vcpkg/files/5629494/config-x64-linux-dbg-err.log)
[config-x64-linux-dbg-out.log](https://github.com/microsoft/vcpkg/files/5629495/config-x64-linux-dbg-out.log)

**Comment by KrasnovPavel** [comment]
On ubuntu it can be fixed by installing libfreeimage-dev from apt. 
`sudo apt install libfreeimage-dev`
For other distros you need to find correct name for development package of freeimage and install it using distro's package manager.

**Comment by dg0yt** [comment]
> On ubuntu it can be fixed by installing libfreeimage-dev from apt.
> `sudo apt install libfreeimage-dev`
> For other distros you need to find correct name for development package of freeimage and install it using distro's package manager.

@KrasnovPavel Better don't mix vcpkg and system libraries unless you understand the implications. The system libraries were not compiled against the same versions and configurations of libraries as the ports in vcpkg. Even if it compiles, it may fail miserably at runtime, due to subtle difference in behaviour.

**Comment by Llcoolsouder** [comment]
> > On ubuntu it can be fixed by installing libfreeimage-dev from apt.
> > `sudo apt install libfreeimage-dev`
> > For other distros you need to find correct name for development package of freeimage and install it using distro's package manager.
> 
> @KrasnovPavel Better don't mix vcpkg and system libraries unless you understand the implications.

@dg0yt If you can't get dependencies through apt, where are you supposed to get them from?

**Comment by dg0yt** [comment]
> If you can't get dependencies through apt, where are you supposed to get them from?

@Llcoolsouder Except for build tools, dependencies should come from vcpkg.

**Comment by Adela0814** [comment]
Fixed in PR #15886
